// cloudflare/resources/ssl/ssl.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as AnalyzeAPI from 'cloudflare/resources/ssl/analyze';
import * as RecommendationsAPI from 'cloudflare/resources/ssl/recommendations';
import * as VerificationAPI from 'cloudflare/resources/ssl/verification';
import * as CertificatePacksAPI from 'cloudflare/resources/ssl/certificate-packs/certificate-packs';
import * as UniversalAPI from 'cloudflare/resources/ssl/universal/universal';
export class SSL extends APIResource {
    constructor() {
        super(...arguments);
        this.analyze = new AnalyzeAPI.Analyze(this._client);
        this.certificatePacks = new CertificatePacksAPI.CertificatePacks(this._client);
        this.recommendations = new RecommendationsAPI.Recommendations(this._client);
        this.universal = new UniversalAPI.Universal(this._client);
        this.verification = new VerificationAPI.VerificationResource(this._client);
    }
}
(function (SSL) {
    SSL.Analyze = AnalyzeAPI.Analyze;
    SSL.CertificatePacks = CertificatePacksAPI.CertificatePacks;
    SSL.CertificatePackListResponsesSinglePage = CertificatePacksAPI.CertificatePackListResponsesSinglePage;
    SSL.Recommendations = RecommendationsAPI.Recommendations;
    SSL.Universal = UniversalAPI.Universal;
    SSL.VerificationResource = VerificationAPI.VerificationResource;
})(SSL || (SSL = {}));
