// cloudflare/resources/workers/workers.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as AccountSettingsAPI from 'cloudflare/resources/workers/account-settings';
import * as AIAPI from 'cloudflare/resources/workers/ai';
import * as DomainsAPI from 'cloudflare/resources/workers/domains';
import * as SubdomainsAPI from 'cloudflare/resources/workers/subdomains';
import * as ScriptsAPI from 'cloudflare/resources/workers/scripts/scripts';
export class Workers extends APIResource {
    constructor() {
        super(...arguments);
        this.ai = new AIAPI.AI(this._client);
        this.scripts = new ScriptsAPI.Scripts(this._client);
        this.accountSettings = new AccountSettingsAPI.AccountSettings(this._client);
        this.domains = new DomainsAPI.Domains(this._client);
        this.subdomains = new SubdomainsAPI.Subdomains(this._client);
    }
}
(function (Workers) {
    Workers.AI = AIAPI.AI;
    Workers.Scripts = ScriptsAPI.Scripts;
    Workers.ScriptsSinglePage = ScriptsAPI.ScriptsSinglePage;
    Workers.AccountSettings = AccountSettingsAPI.AccountSettings;
    Workers.Domains = DomainsAPI.Domains;
    Workers.DomainsSinglePage = DomainsAPI.DomainsSinglePage;
    Workers.Subdomains = SubdomainsAPI.Subdomains;
})(Workers || (Workers = {}));
