// cloudflare/resources/zero-trust/zero-trust.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as ConnectivitySettingsAPI from 'cloudflare/resources/zero-trust/connectivity-settings';
import * as IdentityProvidersAPI from 'cloudflare/resources/zero-trust/identity-providers';
import * as OrganizationsAPI from 'cloudflare/resources/zero-trust/organizations';
import * as SeatsAPI from 'cloudflare/resources/zero-trust/seats';
import * as AccessAPI from 'cloudflare/resources/zero-trust/access/access';
import * as DevicesAPI from 'cloudflare/resources/zero-trust/devices/devices';
import * as DEXAPI from 'cloudflare/resources/zero-trust/dex/dex';
import * as DLPAPI from 'cloudflare/resources/zero-trust/dlp/dlp';
import * as GatewayAPI from 'cloudflare/resources/zero-trust/gateway/gateway';
import * as NetworksAPI from 'cloudflare/resources/zero-trust/networks/networks';
import * as RiskScoringAPI from 'cloudflare/resources/zero-trust/risk-scoring/risk-scoring';
import * as TunnelsAPI from 'cloudflare/resources/zero-trust/tunnels/tunnels';
export class ZeroTrust extends APIResource {
    constructor() {
        super(...arguments);
        this.devices = new DevicesAPI.Devices(this._client);
        this.identityProviders = new IdentityProvidersAPI.IdentityProviders(this._client);
        this.organizations = new OrganizationsAPI.Organizations(this._client);
        this.seats = new SeatsAPI.Seats(this._client);
        this.access = new AccessAPI.Access(this._client);
        this.dex = new DEXAPI.DEX(this._client);
        this.tunnels = new TunnelsAPI.Tunnels(this._client);
        this.connectivitySettings = new ConnectivitySettingsAPI.ConnectivitySettings(this._client);
        this.dlp = new DLPAPI.DLP(this._client);
        this.gateway = new GatewayAPI.Gateway(this._client);
        this.networks = new NetworksAPI.Networks(this._client);
        this.riskScoring = new RiskScoringAPI.RiskScoring(this._client);
    }
}
(function (ZeroTrust) {
    ZeroTrust.Devices = DevicesAPI.Devices;
    ZeroTrust.DevicesSinglePage = DevicesAPI.DevicesSinglePage;
    ZeroTrust.IdentityProviders = IdentityProvidersAPI.IdentityProviders;
    ZeroTrust.IdentityProviderListResponsesSinglePage = IdentityProvidersAPI.IdentityProviderListResponsesSinglePage;
    ZeroTrust.Organizations = OrganizationsAPI.Organizations;
    ZeroTrust.Seats = SeatsAPI.Seats;
    ZeroTrust.Access = AccessAPI.Access;
    ZeroTrust.DEX = DEXAPI.DEX;
    ZeroTrust.Tunnels = TunnelsAPI.Tunnels;
    ZeroTrust.TunnelListResponsesV4PagePaginationArray = TunnelsAPI.TunnelListResponsesV4PagePaginationArray;
    ZeroTrust.ConnectivitySettings = ConnectivitySettingsAPI.ConnectivitySettings;
    ZeroTrust.DLP = DLPAPI.DLP;
    ZeroTrust.Gateway = GatewayAPI.Gateway;
    ZeroTrust.Networks = NetworksAPI.Networks;
    ZeroTrust.RiskScoring = RiskScoringAPI.RiskScoring;
})(ZeroTrust || (ZeroTrust = {}));
