// cloudflare/resources/d1/database.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseListResponsesV4PagePaginationArray = exports.Database = void 0;
const resource_1 = require("cloudflare/resource");
const DatabaseAPI = __importStar(require("cloudflare/resources/d1/database"));
const pagination_1 = require("cloudflare/pagination");
class Database extends resource_1.APIResource {
    /**
     * Returns the created D1 database.
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/d1/database`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns a list of D1 databases.
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/d1/database`, DatabaseListResponsesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes the specified D1 database.
     */
    delete(accountIdentifier, databaseIdentifier, options) {
        return this._client.delete(`/accounts/${accountIdentifier}/d1/database/${databaseIdentifier}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns the specified D1 database.
     */
    get(accountIdentifier, databaseIdentifier, options) {
        return this._client.get(`/accounts/${accountIdentifier}/d1/database/${databaseIdentifier}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Returns the query result.
     */
    query(accountIdentifier, databaseIdentifier, body, options) {
        return this._client.post(`/accounts/${accountIdentifier}/d1/database/${databaseIdentifier}/query`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Database = Database;
class DatabaseListResponsesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.DatabaseListResponsesV4PagePaginationArray = DatabaseListResponsesV4PagePaginationArray;
(function (Database) {
    Database.DatabaseListResponsesV4PagePaginationArray = DatabaseAPI.DatabaseListResponsesV4PagePaginationArray;
})(Database = exports.Database || (exports.Database = {}));
