// cloudflare/resources/radar/http/http.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as SummaryAPI from 'cloudflare/resources/radar/http/summary';
import * as TimeseriesGroupsAPI from 'cloudflare/resources/radar/http/timeseries-groups';
import * as TopAPI from 'cloudflare/resources/radar/http/top';
import * as AsesAPI from 'cloudflare/resources/radar/http/ases/ases';
import * as LocationsAPI from 'cloudflare/resources/radar/http/locations/locations';
export class HTTP extends APIResource {
    constructor() {
        super(...arguments);
        this.top = new TopAPI.Top(this._client);
        this.locations = new LocationsAPI.Locations(this._client);
        this.ases = new AsesAPI.Ases(this._client);
        this.summary = new SummaryAPI.Summary(this._client);
        this.timeseriesGroups = new TimeseriesGroupsAPI.TimeseriesGroups(this._client);
    }
}
(function (HTTP) {
    HTTP.Top = TopAPI.Top;
    HTTP.Locations = LocationsAPI.Locations;
    HTTP.Ases = AsesAPI.Ases;
    HTTP.Summary = SummaryAPI.Summary;
    HTTP.TimeseriesGroups = TimeseriesGroupsAPI.TimeseriesGroups;
})(HTTP || (HTTP = {}));
