// cloudflare/resources/firewall/firewall.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as AccessRulesAPI from 'cloudflare/resources/firewall/access-rules';
import * as LockdownsAPI from 'cloudflare/resources/firewall/lockdowns';
import * as RulesAPI from 'cloudflare/resources/firewall/rules';
import * as UARulesAPI from 'cloudflare/resources/firewall/ua-rules';
import * as WAFAPI from 'cloudflare/resources/firewall/waf/waf';
export class Firewall extends APIResource {
    constructor() {
        super(...arguments);
        this.lockdowns = new LockdownsAPI.Lockdowns(this._client);
        this.rules = new RulesAPI.Rules(this._client);
        this.accessRules = new AccessRulesAPI.AccessRules(this._client);
        this.uaRules = new UARulesAPI.UARules(this._client);
        this.waf = new WAFAPI.WAF(this._client);
    }
}
(function (Firewall) {
    Firewall.Lockdowns = LockdownsAPI.Lockdowns;
    Firewall.LockdownsV4PagePaginationArray = LockdownsAPI.LockdownsV4PagePaginationArray;
    Firewall.Rules = RulesAPI.Rules;
    Firewall.FirewallRulesV4PagePaginationArray = RulesAPI.FirewallRulesV4PagePaginationArray;
    Firewall.AccessRules = AccessRulesAPI.AccessRules;
    Firewall.AccessRuleListResponsesV4PagePaginationArray = AccessRulesAPI.AccessRuleListResponsesV4PagePaginationArray;
    Firewall.UARules = UARulesAPI.UARules;
    Firewall.UARuleListResponsesV4PagePaginationArray = UARulesAPI.UARuleListResponsesV4PagePaginationArray;
    Firewall.WAF = WAFAPI.WAF;
})(Firewall || (Firewall = {}));
