// cloudflare/resources/firewall/rules.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallRulesV4PagePaginationArray = exports.Rules = void 0;
const resource_1 = require("cloudflare/resource");
const core_1 = require("cloudflare/core");
const RulesAPI = __importStar(require("cloudflare/resources/firewall/rules"));
const pagination_1 = require("cloudflare/pagination");
class Rules extends resource_1.APIResource {
    /**
     * Create one or more firewall rules.
     */
    create(zoneIdentifier, body, options) {
        return this._client.post(`/zones/${zoneIdentifier}/firewall/rules`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates an existing firewall rule.
     */
    update(zoneIdentifier, id, body, options) {
        return this._client.put(`/zones/${zoneIdentifier}/firewall/rules/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    list(zoneIdentifier, query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.list(zoneIdentifier, {}, query);
        }
        return this._client.getAPIList(`/zones/${zoneIdentifier}/firewall/rules`, FirewallRulesV4PagePaginationArray, { query, ...options });
    }
    /**
     * Deletes an existing firewall rule.
     */
    delete(zoneIdentifier, id, options) {
        return this._client.delete(`/zones/${zoneIdentifier}/firewall/rules/${id}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates the priority of an existing firewall rule.
     */
    edit(zoneIdentifier, id, body, options) {
        return this._client.patch(`/zones/${zoneIdentifier}/firewall/rules/${id}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches the details of a firewall rule.
     */
    get(zoneIdentifier, params, options) {
        const { path_id, query_id, ...query } = params;
        return this._client.get(`/zones/${zoneIdentifier}/firewall/rules/${path_id}`, {
            query: { id: query_id, ...query },
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Rules = Rules;
class FirewallRulesV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.FirewallRulesV4PagePaginationArray = FirewallRulesV4PagePaginationArray;
(function (Rules) {
    Rules.FirewallRulesV4PagePaginationArray = RulesAPI.FirewallRulesV4PagePaginationArray;
})(Rules = exports.Rules || (exports.Rules = {}));
