// cloudflare/resources/addressing/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Services = exports.ServiceListResponsesSinglePage = exports.Prefixes = exports.PrefixesSinglePage = exports.LOADocuments = exports.Addressing = exports.AddressMaps = exports.AddressMapsSinglePage = void 0;
var index_1 = require("./address-maps/index.js");
Object.defineProperty(exports, "AddressMapsSinglePage", { enumerable: true, get: function () { return index_1.AddressMapsSinglePage; } });
Object.defineProperty(exports, "AddressMaps", { enumerable: true, get: function () { return index_1.AddressMaps; } });
var addressing_1 = require("./addressing.js");
Object.defineProperty(exports, "Addressing", { enumerable: true, get: function () { return addressing_1.Addressing; } });
var index_2 = require("./loa-documents/index.js");
Object.defineProperty(exports, "LOADocuments", { enumerable: true, get: function () { return index_2.LOADocuments; } });
var index_3 = require("./prefixes/index.js");
Object.defineProperty(exports, "PrefixesSinglePage", { enumerable: true, get: function () { return index_3.PrefixesSinglePage; } });
Object.defineProperty(exports, "Prefixes", { enumerable: true, get: function () { return index_3.Prefixes; } });
var services_1 = require("./services.js");
Object.defineProperty(exports, "ServiceListResponsesSinglePage", { enumerable: true, get: function () { return services_1.ServiceListResponsesSinglePage; } });
Object.defineProperty(exports, "Services", { enumerable: true, get: function () { return services_1.Services; } });
