// cloudflare/resources/zero-trust/access/users/users.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as UsersAPI from 'cloudflare/resources/zero-trust/access/users/users';
import * as ActiveSessionsAPI from 'cloudflare/resources/zero-trust/access/users/active-sessions';
import * as FailedLoginsAPI from 'cloudflare/resources/zero-trust/access/users/failed-logins';
import * as LastSeenIdentityAPI from 'cloudflare/resources/zero-trust/access/users/last-seen-identity';
import { SinglePage } from 'cloudflare/pagination';
export class Users extends APIResource {
    constructor() {
        super(...arguments);
        this.activeSessions = new ActiveSessionsAPI.ActiveSessions(this._client);
        this.lastSeenIdentity = new LastSeenIdentityAPI.LastSeenIdentity(this._client);
        this.failedLogins = new FailedLoginsAPI.FailedLogins(this._client);
    }
    /**
     * Gets a list of users for an account.
     */
    list(identifier, options) {
        return this._client.getAPIList(`/accounts/${identifier}/access/users`, AccessUsersSinglePage, options);
    }
}
export class AccessUsersSinglePage extends SinglePage {
}
(function (Users) {
    Users.AccessUsersSinglePage = UsersAPI.AccessUsersSinglePage;
    Users.ActiveSessions = ActiveSessionsAPI.ActiveSessions;
    Users.ActiveSessionListResponsesSinglePage = ActiveSessionsAPI.ActiveSessionListResponsesSinglePage;
    Users.LastSeenIdentity = LastSeenIdentityAPI.LastSeenIdentity;
    Users.FailedLogins = FailedLoginsAPI.FailedLogins;
    Users.FailedLoginListResponsesSinglePage = FailedLoginsAPI.FailedLoginListResponsesSinglePage;
})(Users || (Users = {}));
