// cloudflare/resources/logpush/logpush.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as EdgeAPI from 'cloudflare/resources/logpush/edge';
import * as JobsAPI from 'cloudflare/resources/logpush/jobs';
import * as OwnershipAPI from 'cloudflare/resources/logpush/ownership';
import * as ValidateAPI from 'cloudflare/resources/logpush/validate';
import * as DatasetsAPI from 'cloudflare/resources/logpush/datasets/datasets';
export class Logpush extends APIResource {
    constructor() {
        super(...arguments);
        this.datasets = new DatasetsAPI.Datasets(this._client);
        this.edge = new EdgeAPI.Edge(this._client);
        this.jobs = new JobsAPI.Jobs(this._client);
        this.ownership = new OwnershipAPI.Ownership(this._client);
        this.validate = new ValidateAPI.Validate(this._client);
    }
}
(function (Logpush) {
    Logpush.Datasets = DatasetsAPI.Datasets;
    Logpush.Edge = EdgeAPI.Edge;
    Logpush.Jobs = JobsAPI.Jobs;
    Logpush.LogpushJobsSinglePage = JobsAPI.LogpushJobsSinglePage;
    Logpush.Ownership = OwnershipAPI.Ownership;
    Logpush.Validate = ValidateAPI.Validate;
})(Logpush || (Logpush = {}));
