// cloudflare/resources/secondary-dns/secondary-dns.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as ACLsAPI from 'cloudflare/resources/secondary-dns/acls';
import * as ForceAXFRAPI from 'cloudflare/resources/secondary-dns/force-axfr';
import * as IncomingAPI from 'cloudflare/resources/secondary-dns/incoming';
import * as PeersAPI from 'cloudflare/resources/secondary-dns/peers';
import * as TSIGsAPI from 'cloudflare/resources/secondary-dns/tsigs';
import * as OutgoingAPI from 'cloudflare/resources/secondary-dns/outgoing/outgoing';
export class SecondaryDNS extends APIResource {
    constructor() {
        super(...arguments);
        this.forceAXFR = new ForceAXFRAPI.ForceAXFRResource(this._client);
        this.incoming = new IncomingAPI.IncomingResource(this._client);
        this.outgoing = new OutgoingAPI.OutgoingResource(this._client);
        this.acls = new ACLsAPI.ACLs(this._client);
        this.peers = new PeersAPI.Peers(this._client);
        this.tsigs = new TSIGsAPI.TSIGs(this._client);
    }
}
(function (SecondaryDNS) {
    SecondaryDNS.ForceAXFRResource = ForceAXFRAPI.ForceAXFRResource;
    SecondaryDNS.IncomingResource = IncomingAPI.IncomingResource;
    SecondaryDNS.OutgoingResource = OutgoingAPI.OutgoingResource;
    SecondaryDNS.ACLs = ACLsAPI.ACLs;
    SecondaryDNS.ACLsSinglePage = ACLsAPI.ACLsSinglePage;
    SecondaryDNS.Peers = PeersAPI.Peers;
    SecondaryDNS.PeersSinglePage = PeersAPI.PeersSinglePage;
    SecondaryDNS.TSIGs = TSIGsAPI.TSIGs;
    SecondaryDNS.TSIGsSinglePage = TSIGsAPI.TSIGsSinglePage;
})(SecondaryDNS || (SecondaryDNS = {}));
