// cloudflare/resources/intel/intel.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as DNSAPI from 'cloudflare/resources/intel/dns';
import * as DomainHistoryAPI from 'cloudflare/resources/intel/domain-history';
import * as IPListsAPI from 'cloudflare/resources/intel/ip-lists';
import * as IPsAPI from 'cloudflare/resources/intel/ips';
import * as MiscategorizationsAPI from 'cloudflare/resources/intel/miscategorizations';
import * as SinkholesAPI from 'cloudflare/resources/intel/sinkholes';
import * as WhoisAPI from 'cloudflare/resources/intel/whois';
import * as ASNAPI from 'cloudflare/resources/intel/asn/asn';
import * as AttackSurfaceReportAPI from 'cloudflare/resources/intel/attack-surface-report/attack-surface-report';
import * as DomainsAPI from 'cloudflare/resources/intel/domains/domains';
import * as IndicatorFeedsAPI from 'cloudflare/resources/intel/indicator-feeds/indicator-feeds';
export class Intel extends APIResource {
    constructor() {
        super(...arguments);
        this.asn = new ASNAPI.ASN(this._client);
        this.dns = new DNSAPI.DNS(this._client);
        this.domains = new DomainsAPI.Domains(this._client);
        this.domainHistory = new DomainHistoryAPI.DomainHistoryResource(this._client);
        this.ips = new IPsAPI.IPs(this._client);
        this.ipLists = new IPListsAPI.IPLists(this._client);
        this.miscategorizations = new MiscategorizationsAPI.Miscategorizations(this._client);
        this.whois = new WhoisAPI.Whois(this._client);
        this.indicatorFeeds = new IndicatorFeedsAPI.IndicatorFeeds(this._client);
        this.sinkholes = new SinkholesAPI.Sinkholes(this._client);
        this.attackSurfaceReport = new AttackSurfaceReportAPI.AttackSurfaceReport(this._client);
    }
}
(function (Intel) {
    Intel.ASN = ASNAPI.ASN;
    Intel.DNS = DNSAPI.DNS;
    Intel.DNSListResponsesV4PagePagination = DNSAPI.DNSListResponsesV4PagePagination;
    Intel.Domains = DomainsAPI.Domains;
    Intel.DomainHistoryResource = DomainHistoryAPI.DomainHistoryResource;
    Intel.IPs = IPsAPI.IPs;
    Intel.IPLists = IPListsAPI.IPLists;
    Intel.Miscategorizations = MiscategorizationsAPI.Miscategorizations;
    Intel.Whois = WhoisAPI.Whois;
    Intel.IndicatorFeeds = IndicatorFeedsAPI.IndicatorFeeds;
    Intel.IndicatorFeedListResponsesSinglePage = IndicatorFeedsAPI.IndicatorFeedListResponsesSinglePage;
    Intel.Sinkholes = SinkholesAPI.Sinkholes;
    Intel.SinkholesSinglePage = SinkholesAPI.SinkholesSinglePage;
    Intel.AttackSurfaceReport = AttackSurfaceReportAPI.AttackSurfaceReport;
})(Intel || (Intel = {}));
