// cloudflare/resources/magic-transit/magic-transit.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as CfInterconnectsAPI from 'cloudflare/resources/magic-transit/cf-interconnects';
import * as GRETunnelsAPI from 'cloudflare/resources/magic-transit/gre-tunnels';
import * as IPSECTunnelsAPI from 'cloudflare/resources/magic-transit/ipsec-tunnels';
import * as RoutesAPI from 'cloudflare/resources/magic-transit/routes';
import * as SitesAPI from 'cloudflare/resources/magic-transit/sites/sites';
export class MagicTransit extends APIResource {
    constructor() {
        super(...arguments);
        this.cfInterconnects = new CfInterconnectsAPI.CfInterconnects(this._client);
        this.greTunnels = new GRETunnelsAPI.GRETunnels(this._client);
        this.ipsecTunnels = new IPSECTunnelsAPI.IPSECTunnels(this._client);
        this.routes = new RoutesAPI.Routes(this._client);
        this.sites = new SitesAPI.Sites(this._client);
    }
}
(function (MagicTransit) {
    MagicTransit.CfInterconnects = CfInterconnectsAPI.CfInterconnects;
    MagicTransit.GRETunnels = GRETunnelsAPI.GRETunnels;
    MagicTransit.IPSECTunnels = IPSECTunnelsAPI.IPSECTunnels;
    MagicTransit.Routes = RoutesAPI.Routes;
    MagicTransit.Sites = SitesAPI.Sites;
    MagicTransit.SitesSinglePage = SitesAPI.SitesSinglePage;
})(MagicTransit || (MagicTransit = {}));
