// cloudflare/resources/zero-trust/access/access.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as BookmarksAPI from 'cloudflare/resources/zero-trust/access/bookmarks';
import * as CustomPagesAPI from 'cloudflare/resources/zero-trust/access/custom-pages';
import * as GroupsAPI from 'cloudflare/resources/zero-trust/access/groups';
import * as KeysAPI from 'cloudflare/resources/zero-trust/access/keys';
import * as ServiceTokensAPI from 'cloudflare/resources/zero-trust/access/service-tokens';
import * as TagsAPI from 'cloudflare/resources/zero-trust/access/tags';
import * as ApplicationsAPI from 'cloudflare/resources/zero-trust/access/applications/applications';
import * as CertificatesAPI from 'cloudflare/resources/zero-trust/access/certificates/certificates';
import * as LogsAPI from 'cloudflare/resources/zero-trust/access/logs/logs';
import * as UsersAPI from 'cloudflare/resources/zero-trust/access/users/users';
export class Access extends APIResource {
    constructor() {
        super(...arguments);
        this.applications = new ApplicationsAPI.Applications(this._client);
        this.certificates = new CertificatesAPI.Certificates(this._client);
        this.groups = new GroupsAPI.Groups(this._client);
        this.serviceTokens = new ServiceTokensAPI.ServiceTokens(this._client);
        this.bookmarks = new BookmarksAPI.Bookmarks(this._client);
        this.keys = new KeysAPI.Keys(this._client);
        this.logs = new LogsAPI.Logs(this._client);
        this.users = new UsersAPI.Users(this._client);
        this.customPages = new CustomPagesAPI.CustomPages(this._client);
        this.tags = new TagsAPI.Tags(this._client);
    }
}
(function (Access) {
    Access.Applications = ApplicationsAPI.Applications;
    Access.ApplicationsSinglePage = ApplicationsAPI.ApplicationsSinglePage;
    Access.Certificates = CertificatesAPI.Certificates;
    Access.CertificatesSinglePage = CertificatesAPI.CertificatesSinglePage;
    Access.Groups = GroupsAPI.Groups;
    Access.ZeroTrustGroupsSinglePage = GroupsAPI.ZeroTrustGroupsSinglePage;
    Access.ServiceTokens = ServiceTokensAPI.ServiceTokens;
    Access.ServiceTokensSinglePage = ServiceTokensAPI.ServiceTokensSinglePage;
    Access.Bookmarks = BookmarksAPI.Bookmarks;
    Access.BookmarksSinglePage = BookmarksAPI.BookmarksSinglePage;
    Access.Keys = KeysAPI.Keys;
    Access.Logs = LogsAPI.Logs;
    Access.Users = UsersAPI.Users;
    Access.AccessUsersSinglePage = UsersAPI.AccessUsersSinglePage;
    Access.CustomPages = CustomPagesAPI.CustomPages;
    Access.CustomPageWithoutHTMLsSinglePage = CustomPagesAPI.CustomPageWithoutHTMLsSinglePage;
    Access.Tags = TagsAPI.Tags;
    Access.TagsSinglePage = TagsAPI.TagsSinglePage;
})(Access || (Access = {}));
