// cloudflare/resources/zero-trust/access/bookmarks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookmarksSinglePage = exports.Bookmarks = void 0;
const resource_1 = require("cloudflare/resource");
const BookmarksAPI = __importStar(require("cloudflare/resources/zero-trust/access/bookmarks"));
const pagination_1 = require("cloudflare/pagination");
class Bookmarks extends resource_1.APIResource {
    /**
     * Create a new Bookmark application.
     */
    create(identifier, uuid, body, options) {
        return this._client.post(`/accounts/${identifier}/access/bookmarks/${uuid}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a configured Bookmark application.
     */
    update(identifier, uuid, body, options) {
        return this._client.put(`/accounts/${identifier}/access/bookmarks/${uuid}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists Bookmark applications.
     */
    list(identifier, options) {
        return this._client.getAPIList(`/accounts/${identifier}/access/bookmarks`, BookmarksSinglePage, options);
    }
    /**
     * Deletes a Bookmark application.
     */
    delete(identifier, uuid, options) {
        return this._client.delete(`/accounts/${identifier}/access/bookmarks/${uuid}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a single Bookmark application.
     */
    get(identifier, uuid, options) {
        return this._client.get(`/accounts/${identifier}/access/bookmarks/${uuid}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Bookmarks = Bookmarks;
class BookmarksSinglePage extends pagination_1.SinglePage {
}
exports.BookmarksSinglePage = BookmarksSinglePage;
(function (Bookmarks) {
    Bookmarks.BookmarksSinglePage = BookmarksAPI.BookmarksSinglePage;
})(Bookmarks = exports.Bookmarks || (exports.Bookmarks = {}));
