// cloudflare/resources/zero-trust/access/custom-pages.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomPageWithoutHTMLsSinglePage = exports.CustomPages = void 0;
const resource_1 = require("cloudflare/resource");
const CustomPagesAPI = __importStar(require("cloudflare/resources/zero-trust/access/custom-pages"));
const pagination_1 = require("cloudflare/pagination");
class CustomPages extends resource_1.APIResource {
    /**
     * Create a custom page
     */
    create(identifier, body, options) {
        return this._client.post(`/accounts/${identifier}/access/custom_pages`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update a custom page
     */
    update(identifier, uuid, body, options) {
        return this._client.put(`/accounts/${identifier}/access/custom_pages/${uuid}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List custom pages
     */
    list(identifier, options) {
        return this._client.getAPIList(`/accounts/${identifier}/access/custom_pages`, CustomPageWithoutHTMLsSinglePage, options);
    }
    /**
     * Delete a custom page
     */
    delete(identifier, uuid, options) {
        return this._client.delete(`/accounts/${identifier}/access/custom_pages/${uuid}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a custom page and also returns its HTML.
     */
    get(identifier, uuid, options) {
        return this._client.get(`/accounts/${identifier}/access/custom_pages/${uuid}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.CustomPages = CustomPages;
class CustomPageWithoutHTMLsSinglePage extends pagination_1.SinglePage {
}
exports.CustomPageWithoutHTMLsSinglePage = CustomPageWithoutHTMLsSinglePage;
(function (CustomPages) {
    CustomPages.CustomPageWithoutHTMLsSinglePage = CustomPagesAPI.CustomPageWithoutHTMLsSinglePage;
})(CustomPages = exports.CustomPages || (exports.CustomPages = {}));
