// cloudflare/resources/zero-trust/access/tags.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagsSinglePage = exports.Tags = void 0;
const resource_1 = require("cloudflare/resource");
const TagsAPI = __importStar(require("cloudflare/resources/zero-trust/access/tags"));
const pagination_1 = require("cloudflare/pagination");
class Tags extends resource_1.APIResource {
    /**
     * Create a tag
     */
    create(identifier, body, options) {
        return this._client.post(`/accounts/${identifier}/access/tags`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Update a tag
     */
    update(identifier, tagName, body, options) {
        return this._client.put(`/accounts/${identifier}/access/tags/${tagName}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List tags
     */
    list(identifier, options) {
        return this._client.getAPIList(`/accounts/${identifier}/access/tags`, TagsSinglePage, options);
    }
    /**
     * Delete a tag
     */
    delete(identifier, name, options) {
        return this._client.delete(`/accounts/${identifier}/access/tags/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get a tag
     */
    get(identifier, name, options) {
        return this._client.get(`/accounts/${identifier}/access/tags/${name}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Tags = Tags;
class TagsSinglePage extends pagination_1.SinglePage {
}
exports.TagsSinglePage = TagsSinglePage;
(function (Tags) {
    Tags.TagsSinglePage = TagsAPI.TagsSinglePage;
})(Tags = exports.Tags || (exports.Tags = {}));
