// cloudflare/resources/zero-trust/dex/dex.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from 'cloudflare/resource';
import * as ColosAPI from 'cloudflare/resources/zero-trust/dex/colos';
import * as TracerouteTestsAPI from 'cloudflare/resources/zero-trust/dex/traceroute-tests';
import * as FleetStatusAPI from 'cloudflare/resources/zero-trust/dex/fleet-status/fleet-status';
import * as HTTPTestsAPI from 'cloudflare/resources/zero-trust/dex/http-tests/http-tests';
import * as TestsAPI from 'cloudflare/resources/zero-trust/dex/tests/tests';
import * as TracerouteTestResultsAPI from 'cloudflare/resources/zero-trust/dex/traceroute-test-results/traceroute-test-results';
export class DEX extends APIResource {
    constructor() {
        super(...arguments);
        this.colos = new ColosAPI.Colos(this._client);
        this.fleetStatus = new FleetStatusAPI.FleetStatus(this._client);
        this.httpTests = new HTTPTestsAPI.HTTPTests(this._client);
        this.tests = new TestsAPI.Tests(this._client);
        this.tracerouteTestResults = new TracerouteTestResultsAPI.TracerouteTestResults(this._client);
        this.tracerouteTests = new TracerouteTestsAPI.TracerouteTests(this._client);
    }
}
(function (DEX) {
    DEX.Colos = ColosAPI.Colos;
    DEX.ColoListResponsesSinglePage = ColosAPI.ColoListResponsesSinglePage;
    DEX.FleetStatus = FleetStatusAPI.FleetStatus;
    DEX.HTTPTests = HTTPTestsAPI.HTTPTests;
    DEX.Tests = TestsAPI.Tests;
    DEX.TestListResponsesV4PagePagination = TestsAPI.TestListResponsesV4PagePagination;
    DEX.TracerouteTestResults = TracerouteTestResultsAPI.TracerouteTestResults;
    DEX.TracerouteTests = TracerouteTestsAPI.TracerouteTests;
})(DEX || (DEX = {}));
