// cloudflare/resources/zero-trust/dex/dex.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEX = void 0;
const resource_1 = require("cloudflare/resource");
const ColosAPI = __importStar(require("cloudflare/resources/zero-trust/dex/colos"));
const TracerouteTestsAPI = __importStar(require("cloudflare/resources/zero-trust/dex/traceroute-tests"));
const FleetStatusAPI = __importStar(require("cloudflare/resources/zero-trust/dex/fleet-status/fleet-status"));
const HTTPTestsAPI = __importStar(require("cloudflare/resources/zero-trust/dex/http-tests/http-tests"));
const TestsAPI = __importStar(require("cloudflare/resources/zero-trust/dex/tests/tests"));
const TracerouteTestResultsAPI = __importStar(require("cloudflare/resources/zero-trust/dex/traceroute-test-results/traceroute-test-results"));
class DEX extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.colos = new ColosAPI.Colos(this._client);
        this.fleetStatus = new FleetStatusAPI.FleetStatus(this._client);
        this.httpTests = new HTTPTestsAPI.HTTPTests(this._client);
        this.tests = new TestsAPI.Tests(this._client);
        this.tracerouteTestResults = new TracerouteTestResultsAPI.TracerouteTestResults(this._client);
        this.tracerouteTests = new TracerouteTestsAPI.TracerouteTests(this._client);
    }
}
exports.DEX = DEX;
(function (DEX) {
    DEX.Colos = ColosAPI.Colos;
    DEX.ColoListResponsesSinglePage = ColosAPI.ColoListResponsesSinglePage;
    DEX.FleetStatus = FleetStatusAPI.FleetStatus;
    DEX.HTTPTests = HTTPTestsAPI.HTTPTests;
    DEX.Tests = TestsAPI.Tests;
    DEX.TestListResponsesV4PagePagination = TestsAPI.TestListResponsesV4PagePagination;
    DEX.TracerouteTestResults = TracerouteTestResultsAPI.TracerouteTestResults;
    DEX.TracerouteTests = TracerouteTestsAPI.TracerouteTests;
})(DEX = exports.DEX || (exports.DEX = {}));
